aten/src/ATen/ThreadLocalState.cpp

```diff
@@ -29,6 +33,8 @@ void ThreadLocalState::setThreadLocalState(
 
   _setTLSCallbacks(state.callbacks_);
 
+  at::enableRecordFunction(state.observers_enabled_);
```

Comment:
just curious - why do you put it separate and don't use threadlocaldebuginfo and those setters/getters you had

Reply:
ThreadLocalDebugInfo is for range (scope) based settings and this one is per thread, and having it close to the definition of record function seemed more reasonable